In [1]:
import numpy as np # linear algebra 
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)
# stats models: regression fitting via formulas
import statsmodels.formula.api as smf
# stats models: regression fitting via matrices of regression design
import statsmodels.api as sm

In [2]:
df = pd.read_csv('https://raw.githubusercontent.com/artamonoff/Econometrica/master/python-notebooks/data-csv/sleep75.csv')
df[['sleep', 'hrwage', 'smsa', 'totwrk', 'male', 'marr', 'age', 'south', 'yngkid']].describe()

sleep      hrwage        smsa       totwrk        male  \
count   706.000000  532.000000  706.000000   706.000000  706.000000   
mean   3266.355524    5.082839    0.399433  2122.920680    0.566572   
std     444.413448    3.704385    0.490129   947.470123    0.495900   
min     755.000000    0.350000    0.000000     0.000000    0.000000   
25%    3015.000000    2.890002    0.000000  1553.500000    0.000000   
50%    3270.500000    4.379999    0.000000  2288.000000    1.000000   
75%    3532.250000    6.210001    1.000000  2691.750000    1.000000   
max    4695.000000   35.509987    1.000000  6415.000000    1.000000   

            marr         age       south      yngkid  
count  706.00000  706.000000  706.000000  706.000000  
mean     0.82153   38.815864    0.184136    0.128895  
std      0.38318   11.342637    0.387870    0.335321  
min      0.00000   23.000000    0.000000    0.000000  
25%      1.00000   29.000000    0.000000    0.000000  
50%      1.00000   36.000000    0.000000    0.000000  
75%      1.00000   48.000000    0.000000    0.000000  
max      1.00000   65.000000    1.000000    1.000000

In [3]:
# специфицируем моделл через формулу
sleep_eq1 = smf.ols(formula='sleep~np.log(hrwage)+smsa+totwrk+male+marr+age+south+yngkid', data=df).fit()
# Коэфициенты модели с округление
sleep_eq1.params.round(2)

Intercept         3440.19
np.log(hrwage)      -1.39
smsa               -36.96
totwrk              -0.16
male                36.87
marr                53.34
age                  2.37
south               76.27
yngkid              47.92
dtype: float64

### Спецификация 
$$
sleep=3440.19-1.39*np.log(hrwage)-36.96*smsa-0.16*totwrk+ 36.87*male+53.34*marr+2.37*age+76.27*south+47.92*yngkid
$$

In [5]:
# Число наблюдений, под которые подгонялась модель
sleep_eq1.nobs

532.0

### Интерпретация
1. При учеличении почасовой оплаты на 1%, количество минут сна в неделю уменьшится на 0.00139 минут
2. Жители мегаполисов спят на 39.96 минут меньше, чем те, кто не живут в мегаполисе, при прочих равных условиях.